### PR TITLE
feat(MintLimiter): gateway integration

### PR DESCRIPTION
**Pull request by re1ro**
* [x] `MintLimiter`: Gateway integration
* [ ] Depends on #198

### CODE REVIEW

File: contracts/AxelarGateway.sol [hunk 1]
```diff
@@ -52,19 +52,24 @@ contract AxelarGateway is IAxelarGateway, EternalStorage {
...
 
     constructor(
         address authModule_,
         address governanceModule_,
+        address mintLimitModule_,
         address tokenDeployerImplementation_
     ) {
         if (authModule_.code.length == 0) revert InvalidAuthModule();
         if (governanceModule_.code.length == 0) revert InvalidGovernanceModule();
+        if (mintLimitModule_.code.length == 0) revert InvalidMintLimiterModule();
```
Comment (milapsheth) [inline contracts/AxelarGateway.sol:67]: similar comment about not requiring this to be a contract

File: contracts/AxelarGateway.sol [hunk 2]
```diff
@@ -178,6 +189,10 @@ contract AxelarGateway is IAxelarGateway, EternalStorage {
         return GOVERNANCE_MODULE;
     }
 
+    function mintLimiterModule() public view returns (address) {
+        return MINT_LIMIT_MODULE;
```
Comment (milapsheth) [inline contracts/AxelarGateway.sol:193]: should allow updating this in setup

Comment (milapsheth) [reply contracts/AxelarGateway.sol:193]: nvm, will be done by upgrading implementation since it's an immutable

**Review by Foivos — COMMENTED** [review]
Ι agree that the IAxelarGateway should not be changed for now, but this is happening in a future PR.

**Pull request merged**